File: app/src/main/java/com/artear/multitrackerandroid/trackers/type/exception/MyException.java
package com.artear.multitrackerandroid.trackers.type.exception;

import com.artear.multitracker.contract.send.TrackerException;


public class MyException implements TrackerException {

    public enum ErrorCode {
        INVALID_CHARACTER,
        INTERNAL_ERROR
    }

    private ErrorCode code;
    private String messenger;

    public MyException(ErrorCode code, String messenger) {
        this.code = code;
        this.messenger = messenger;
    }

    @Override
    public String toString() {
        return "< MyException - code = " + code + ", messenger = " + messenger + " >";
    }
}

File: app/src/main/java/com/artear/multitrackerandroid/trackers/type/exception/MyException.kt
package com.artear.multitrackerandroid.trackers.type.exception

import com.artear.multitracker.contract.send.TrackerException


class MyException(private val code: ErrorCode, private val messenger: String) : TrackerException {

    enum class ErrorCode {
        INVALID_CHARACTER,
        INTERNAL_ERROR
    }

    override fun toString(): String {
        return "< MyException - code = $code, messenger = $messenger >"
    }
}
